### DISCUSSION TITLE
(Looking for feedback!) Rewrite to be serverless?

### DISCUSSION THREAD

**Discussion by Selbi182** [Selbi182/SpotifyBigPicture#22] [Announcements]
One of the biggest complaints I have received over time is the hassle that is having to run a Java application in the background for this app to work. It's clunky, it's needlessly complicated, and it's annoying to have an extra terminal open at all times.

There were two reasons for this decision:

1. I programmed it like this because, from the start, I planned to have the server side do all the annoying data hassling with Spotify's API, while the frontend side should only receive ready-to-display data. The browser is already having enough to deal with by rendering everything anyway.
2. Without a server, [remote control](https://github.com/Selbi182/SpotifyBigPicture/blob/master/SETTINGS.md#remote-control) would be impossible. Since I use the app on a hard-to-access Raspberry Pi connected to a TV, I want to deal with it as little as possible. But I also know that many people just want to use it on their desktops and don't care for any remote-controlling options.

But as it stands, setting up the app is still a major pain in the ass for most users, and it's practically impossible to use on a mobile phone without a private webserver to host the app on.

What is your stance on this? Would you like to see a browser-side-only, Java-less rewrite? Or do you prefer to keep it the way it is?

**Comment by Aaantoni** [comment]
Currently I'm only using the app on my desktop to display a floating window on my second monitor, but I do want to get a RPi to connect to my TV for a variety of reasons, the app being one. Some kind of a hybrid solution would be ideal but I don't know how to approach it, maybe split the release into two? Make a .exe for Windows that you can install locally into like AppData that just runs the server locally in the background which you can then either close from inside the browser app itself or from the taskbar; and a separate remote-server-focused build intended for RPi and the likes. But I'm just spitballing here, I don't have any actual experience with any of the stuff you're dealing with here lmao

**Comment by Selbi182** [comment]
Hmm, the idea of separating the logic could be a good idea. It would certainly help reduce the file size a lot (most of the roughly 20 MB file size is the result of SpringBoot Java dependencies). I'll have to think about this for a bit, cause it'd be a major rewrite one way or another.

Thanks for your input!
